OpenRA.Mods.Common/Widgets/SupportPowerTimerWidget.cs

```diff
@@ -21,45 +21,137 @@ public class SupportPowerTimerWidget : Widget
 	{
 		public readonly string Font = "Bold";
 		public readonly string Format = "{0}: {1}";
-		public readonly TimerOrder Order = TimerOrder.Descending;
+		[Translate] public readonly string HeaderAllies = "Allies";
+		[Translate] public readonly string HeaderEnemies = "Enemies";
+		public enum TimerOrder { AlliesFirst, EnemiesFirst }
+		public readonly TimerOrder Order = TimerOrder.EnemiesFirst;
 
 		readonly IEnumerable<SupportPowerInstance> powers;
-		Pair<string, Color>[] texts;
+		readonly IEnumerable<SupportPowerInstance> powersAllies;
+		readonly IEnumerable<SupportPowerInstance> powersEnemies;
+		readonly IEnumerable<Player> playersAllies;
+		readonly IEnumerable<Player> playersEnemies;
+		readonly IEnumerable<Player> allPlayers;
+		readonly bool showGrouping;
+		Pair<string, Color>[] textsAllies;
+		Pair<string, Color>[] textsEnemies;
+		Pair<string, Color>[] allTexts;
 
 		[ObjectCreator.UseCtor]
 		public SupportPowerTimerWidget(World world)
 		{
 			powers = world.ActorsWithTrait<SupportPowerManager>()
-				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant)
+				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant && p.Actor.Owner != world.RenderPlayer)
 				.SelectMany(s => s.Trait.Powers.Values)
 				.Where(p => p.Instances.Any() && p.Info.DisplayTimer && !p.Disabled);
+
+			if (world.RenderPlayer == null)
```

Comment:
guess so it doesnt render the normal UI? I think Local Player changes when you select different players in a replay.